src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1051,7 +1056,10 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 				this._terminalService.focusInstance(terminal);
 			}
 		}
-		this._activeTasks[task.getMapKey()].terminal = terminal;
+		if (this._activeTasks[task.getMapKey()]) {
```

Comment:
Can the types for these string dictionaries be fixed to catch this at compile time? Something like this?

```
private _activeTasks: IStringDictionary<IActiveTerminalData | undefined>;
private _busyTasks: IStringDictionary<Task | undefined>;
private _terminals: IStringDictionary<ITerminalData | undefined>;
```

Reply:
I think we always want to remove the task's map key vs keeping it in the map with a value of `undefined`?

Reply:
You may attempt to keep it in sync, but it's possible to get undefined so the compiler should know that and warn us if we may be causing an NPE

Reply:
Approved, but this is a good debt task which could catch some errors